### PR TITLE
Add options of thumbs position, direction and quantity per row.

### PR DESCRIPTION
**Pull request by giovanapereira**
#### What is the purpose of this pull request?
The purpose of this pull request is add position options of thumbs and main image of product (bottom and top layout) and options to custom direction and quantity of thumbs per row (direction and slidePerView)

#### What problem is this solving?
Doesn't exist a option to edit the position, if exist doesn't work. I try and create a issue, but nobody awnser me, so I resolve on my own. ([Link here](https://github.com/vtex-apps/product-details/issues/127))

#### How should this be manually tested?
https://dev--txboot.myvtex.com/men-s-anderson-bean-electric-blue-black-elephant/p

#### Screenshots or example usage
image

image

image

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ProductImages/components/Carousel/index.js [hunk 1]
```diff
@@ -266,8 +267,8 @@ class Carousel extends Component {
       watchSlidesVisibility: true,
       watchSlidesProgress: true,
       freeMode: true,
-      direction: 'vertical',
-      slidesPerView: 'auto',
+      direction: direction,
+      slidesPerView: slidePerView,
```
Comment (jgfidelis) [inline react/components/ProductImages/components/Carousel/index.js]:
```suggestion
      slidesPerView,
```

File: react/components/ProductImages/components/Carousel/index.js [hunk 2]
```diff
@@ -266,8 +267,8 @@ class Carousel extends Component {
       watchSlidesVisibility: true,
       watchSlidesProgress: true,
       freeMode: true,
-      direction: 'vertical',
-      slidesPerView: 'auto',
+      direction: direction,
```
Comment (jgfidelis) [inline react/components/ProductImages/components/Carousel/index.js]:
```suggestion
      direction,
```

File: react/components/ProductImages/index.js [hunk 3]
```diff
@@ -126,4 +127,59 @@ ProductImages.getSchema = ({ zoomProps: { zoomType } = {} }) => {
...
+            default: 'vertical',
+          },
+          ...(direction),
+        },
+      },
+    },
+  }
+}
+
+ProductImages.getSchema = ({ thumb: { slidePerView } = {} }) => {
```
Comment (jgfidelis) [inline react/components/ProductImages/index.js]: Why getSchema again here?

And the schema should be in the Wrapper component

File: react/components/ProductImages/components/Carousel/index.js [hunk 4]
```diff
@@ -287,20 +288,30 @@ class Carousel extends Component {
       {
         'ml-20-ns w-80-ns': position === 'left' && slides.length > 1,
         'mr-20-ns w-80-ns': position === 'right' && slides.length > 1,
+        'w-100': position === 'top' && slides.length > 1,
+        'w-100': position === 'bottom' && slides.length > 1
```
Comment (lbebber) [inline react/components/ProductImages/components/Carousel/index.js]: verificou se o `w-100` é aplicado nos dois casos? isso acaba sendo um objeto com duas chaves iguais ("`w-100`"), e acho que só vai valer a ultima

File: react/components/ProductImages/components/Carousel/index.js [hunk 5]
```diff
@@ -287,20 +288,30 @@ class Carousel extends Component {
...
-      `w-20 ${styles.carouselGaleryThumbs} bottom-0 top-0 absolute dn`,
+      `w-20 ${styles.carouselGaleryThumbs} dn`,
       {
         'db-ns': slides.length > 1,
-        'left-0 pr5': position === 'left',
-        'right-0 pl5': position === 'right',
+        'absolute bottom-0 top-0 left-0 pr5': position === 'left',
+        'absolute bottom-0 top-0 right-0 pl5': position === 'right',
+        'w-100 pb5': position === 'top',
+        'w-100 pt5': position === 'bottom'
```
Comment (lbebber) [inline react/components/ProductImages/components/Carousel/index.js]: idem

File: react/components/ProductImages/components/Carousel/index.js [hunk 6]
```diff
@@ -287,20 +288,30 @@ class Carousel extends Component {
...
+      }
+    )
+
+    const thumbReverse = classNames(
```
Comment (lbebber) [inline react/components/ProductImages/components/Carousel/index.js]: `thumbReverseClasses`

File: react/components/ProductImages/index.js [hunk 7]
```diff
@@ -126,4 +127,59 @@ ProductImages.getSchema = ({ zoomProps: { zoomType } = {} }) => {
   }
 }
 
+ProductImages.getSchema = ({ thumb: { direction } = {} }) => {
```
Comment (lbebber) [inline react/components/ProductImages/index.js]: como esses schemas sao de layout, precisa colocar `isLayout: true` nas propriedades

**Comment by klzns** [comment]
@allcontributors add @giovanapereira for code

File: react/components/ProductImages/components/Carousel/index.js (hunk 4 above)
Comment (jgfidelis) [reply react/components/ProductImages/components/Carousel/index.js]: poderia ser (position === top || position === bottom) && slides.length > 1

File: react/components/ProductImages/components/Carousel/index.js (hunk 5 above)
Comment (jgfidelis) [reply react/components/ProductImages/components/Carousel/index.js]: juntar numa cond, position === top || posiiton === bottom

Comment (giovanapereira) [reply react/components/ProductImages/components/Carousel/index.js]: Aqui o padrão é w-20 e se vcs notarem existem a classe de padding diferente para bottom e top, então pretendo permanecer desta forma, ok ?

File: react/components/ProductImages/Wrapper.js [hunk 8]
```diff
@@ -40,6 +49,7 @@ ProductImagesWrapper.schema = {
   title: 'admin/editor.product-details.title',
   description: 'admin/editor.product-details.description',
   type: 'object',
+  isLayout: true,
```
Comment (lbebber) [inline react/components/ProductImages/Wrapper.js]: ah, quis dizer colocar `isLayout` em `direction` e em `slidesPerView` (assim como já tem em `thumbnailPosition`)

**Comment by klzns** [comment]
Made some adjustments like changelog, readme, translations and rebased the branch (beware, I did a push force in your branch)

I linked in my workspace with the properties 'top' and 'vertical' and it rendered this:

image

**Comment by giovanapereira** [comment]
> Made some adjustments like changelog, readme, translations and rebased the branch (beware, I did a push force in your branch)
> 
> I linked in my workspace with the properties 'top' and 'vertical' and it rendered this:
> 
> image
> 
> I don't have time to tackle this right now, but this needs to be fixed before we merge this.

Can we just insert a if when the user choose position to verify the direction of thumbs ? I mean, when the position was top or bottom the direction always be horizontal and left or right always be vertical ? So we just remove the option 'direction' of the blocks.

**Comment by klzns** [comment]
Fixed at https://github.com/vtex-apps/store-components/pull/542

**Pull request closed without merging**